bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -822,10 +831,10 @@ public void run()
       };
     }
 
-    protected void release(boolean wait)
+    protected void release(boolean wait, boolean writer)
     {
       final int refCount = this.refCount.decrementAndGet();
-      if (refCount == 0 && storage != null) {
+      if (refCount == 0 && storage != null && canRelease(writer)) {
```

Comment:
I don't think you can make a blanket statement like that. DataList for practical purposes is a queue and queue implementations enforce blocking all the time, think ArrayBlockingQueue.
 
While there may be opportunity to implement more advanced flow control with the queues inside the queue or using pluggable mechanism, this implementation aims to provide a default control mechanism where none is present today.